forgerock-auth/src/androidTest/java/org/forgerock/android/auth/callback/KeyAttestationTest.java

```diff
@@ -0,0 +1,554 @@
+/*
+ * Copyright (c) 2022 - 2023 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.auth.callback;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import android.os.Build;
+
+import androidx.test.core.app.ActivityScenario;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import com.nimbusds.jose.util.Base64;
+import com.nimbusds.jwt.JWT;
+import com.nimbusds.jwt.JWTParser;
+import com.nimbusds.jwt.SignedJWT;
+
+import org.assertj.core.api.Assertions;
+import org.forgerock.android.auth.DummyActivity;
+import org.forgerock.android.auth.FRListener;
+import org.forgerock.android.auth.FRSession;
+import org.forgerock.android.auth.InitProvider;
+import org.forgerock.android.auth.Node;
+import org.forgerock.android.auth.NodeListener;
+import org.forgerock.android.auth.NodeListenerFuture;
+import org.forgerock.android.auth.devicebind.ApplicationPinDeviceAuthenticator;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.text.ParseException;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+@RunWith(AndroidJUnit4.class)
+public class KeyAttestationTest extends BaseDeviceBindingTest {
+    protected final static String TREE = "key-attestation";
+
+    @Test
+    public void testKeyAttestationNoneNone() throws ExecutionException, InterruptedException {
```

Comment:
Is this a typo? Should it be testKeyAttestationNone()

Reply:
I named it "NoneNone", because of the 2 simultaneous possible configurations/permutations - i.e. "NONE" for "AuthenticationType" and "NONE" for the "Key Attestation" property... 
But I agree that this doesn't seem to make much sense... so I'm going to update the names...

Reply:
Got it.